### PR TITLE
modules: Zcbor

### PR DESCRIPTION
**Pull request by de-nordic**
The commit adds zcbor module.

### CODE REVIEW

**Review by tejlmand — COMMENTED** [review]
minor observations

File: west.yml [hunk 1]
```diff
@@ -237,6 +239,10 @@ manifest:
       path: modules/tee/tf-m/psa-arch-tests
       groups:
         - tee
+    - name: zcbor
+      remote: nordic
```
[the remark this answers is not in the document]
Comment (de-nordic) [reply west.yml:243]: Yes, but that requires https://github.com/zephyrproject-rtos/zephyr/issues/43333 to go first.

**Review by nordicjm — DISMISSED** [review]

File: modules/zcbor/Kconfig [hunk 2]
```diff
@@ -0,0 +1,11 @@
+# Copyright (c) 2022 Nordic Semiconductor ASA
+# SPDX-License-Identifier: Apache-2.0
+
+config ZEPHYR_ZCBOR_MODULE
+	bool
+
+config ZCBOR
+	bool "zcbor library"
+	depends on !MINIMAL_LIBC
```
Comment (carlescufi) [inline modules/zcbor/Kconfig:9]: We have to be able to use both minimal libc and newlib with zcbor. The fix here would be to add the required ranges in minimal libc.

**Review by carlescufi — CHANGES_REQUESTED** [review]
NACKing until the C library issue is resolved

File: modules/zcbor/Kconfig (hunk 2 above)
Comment (de-nordic) [reply modules/zcbor/Kconfig]: Removed

**Review by carlescufi — APPROVED** [review]

**Comment by carlescufi** [comment]
@lairdjm can you give a +1 so we can then continue in #43878?

File: west.yml [hunk 3]
```diff
@@ -237,6 +237,9 @@ manifest:
       path: modules/tee/tf-m/psa-arch-tests
       groups:
         - tee
+    - name: zcbor
+      revision: 317d3c6cd6dee23a151a3e7982d8564e7e19b2e2
```
Comment (oyvindronningstad) [inline west.yml:241]: Could be 0.4.0 now.

**Review by oyvindronningstad — APPROVED** [review]

**Review by nordicjm — APPROVED** [review]

File: modules/zcbor/Kconfig [hunk 4]
```diff
@@ -0,0 +1,10 @@
+# Copyright (c) 2022 Nordic Semiconductor ASA
+# SPDX-License-Identifier: Apache-2.0
+
+config ZEPHYR_ZCBOR_MODULE
+	bool
+
+config ZCBOR
+	bool "zcbor library"
+	help
+	  Enable zcbor CBOR encoder/decoder library
```
[the remark this answers is not in the document]
Comment (carlescufi) [reply modules/zcbor/Kconfig:10]: I think we can do that, yes. I wonder however if it'd be better to just have that Kconfig file in the zcbor repo and then `orsource` it from nrf/zephyr?

[the remark this answers is not in the document]
Comment (carlescufi) [reply modules/zcbor/Kconfig:10]: See https://github.com/zephyrproject-rtos/zephyr/pull/44949

**Pull request merged**